maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/extensions/EventConsumerThreadTest.java

```diff
@@ -0,0 +1,492 @@
+    {
+        ForkedProcessEventType[] enums = ForkedProcessEventType.values();
+        TreeMap<Integer, ForkedProcessEventType> map = new TreeMap<>();
+        for (ForkedProcessEventType e : enums) {
+            map.put( e.getOpcode().hashCode(), e );
+        }
+        TimeUnit.SECONDS.sleep( 2 );
+        System.gc();
+        TimeUnit.SECONDS.sleep( 5 );
+        ForkedProcessEventType s = null;
+        int hash = ForkedProcessEventType.BOOTERCODE_STDOUT.hashCode();
+        long l1 = System.currentTimeMillis();
+        for (int i = 0; i < 10_000_000; i++) {
+            s = map.get( hash );
+        }
+        long l2 = System.currentTimeMillis();
+        System.out.println(l2 - l1);
+        System.out.println(s);
+    }
+
+    @Test
+    public void test10() throws InterruptedException
+    {
+        ForkedProcessEventType[] enums = ForkedProcessEventType.values();
+        Map<Segment, ForkedProcessEventType> map = new HashMap<>();
+        for (ForkedProcessEventType e : enums) {
+            byte[] array = e.getOpcode().getBytes();
+            map.put( new Segment( array, 0, array.length ), e );
+        }
+        TimeUnit.SECONDS.sleep( 2 );
+        System.gc();
+        TimeUnit.SECONDS.sleep( 5 );
+        ForkedProcessEventType s = null;
+        byte[] array = ForkedProcessEventType.BOOTERCODE_STDOUT.getOpcode().getBytes();
+        Segment segment = new Segment( array, 0, array.length );
+        long l1 = System.currentTimeMillis();
+        for (int i = 0; i < 10_000_000; i++) {
+            s = map.get( new Segment( array, 0, array.length ) ); // 33.7 nanos
+        }
+        long l2 = System.currentTimeMillis();
+        System.out.println(l2 - l1);
+        System.out.println(s);
+    }
+
+    @Test
+    public void test11() throws InterruptedException
+    {
+        CharsetDecoder decoder = STREAM_ENCODING.newDecoder()
+            .onMalformedInput( REPLACE )
+            .onUnmappableCharacter( REPLACE );
+        ByteBuffer buffer = ByteBuffer.wrap( ForkedProcessEventType.BOOTERCODE_STDOUT.getOpcode().getBytes( UTF_8 ) );
+        CharBuffer chars = CharBuffer.allocate( 100 );
+        TimeUnit.SECONDS.sleep( 2 );
+        System.gc();
+        TimeUnit.SECONDS.sleep( 5 );
+        long l1 = System.currentTimeMillis();
+        for ( int i = 0; i < 10_000_000; i++ )
+        {
+            decoder
+                .reset()
+                .decode( buffer, chars, true );
+            String s = chars.flip().toString(); // 37 nanos = CharsetDecoder + toString
+            buffer.clear();
+            chars.clear();
+            ForkedProcessEventType.byOpcode( s ); // 65 nanos = CharsetDecoder + toString + byOpcode
+        }
+        long l2 = System.currentTimeMillis();
+        System.out.println(l2 - l1);
+    }
+
+    @Test
+    public void test12() throws InterruptedException
+    {
+        StringBuilder builder = new StringBuilder( 256 );
+        TimeUnit.SECONDS.sleep( 2 );
+        System.gc();
```

Comment:
What is the purpose of this?

Reply:
This was deleted.
Pls have a look at the commits. There are improvements in `EventConsumerThread`.
I have added the tests in `EventConsumerThread` with 47% coverage. I continue with testing it more...